packages/expo-updates/ios/EXUpdates/AppLoader/EXUpdatesFileDownloader.m

```diff
@@ -84,50 +84,49 @@ - (void)downloadManifestFromURL:(NSURL *)url
                                                      timeoutInterval:EXUpdatesDefaultTimeoutInterval];
   [self _setManifestHTTPHeaderFields:request];
   [self _downloadDataWithRequest:request successBlock:^(NSData *data, NSURLResponse *response) {
+    NSAssert([response isKindOfClass:[NSHTTPURLResponse class]], @"response must be a NSHTTPURLResponse");
```

Comment:
The NSAssert calls are no-ops in production builds, so I've been (slowly) trying to move away from them in new code -- would prefer actual logic to handle the bad case at runtime and avoid a crash.

(If we do somehow get a mistyped object at runtime, it will crash if we e.g. try to call an NSString method on an object that isn't an NSString.)